src/System.Management.Automation/utils/PathUtils.cs

```diff
@@ -470,6 +489,9 @@ internal static Encoding Convert(Cmdlet cmdlet, string encoding)
             if (string.Equals(encoding, String, StringComparison.OrdinalIgnoreCase))
                 return System.Text.Encoding.Unicode;
 
+            if (string.Equals(encoding, Byte, StringComparison.OrdinalIgnoreCase))
+                return byteEncoding;
```

Comment:
If we aren't using a dictionary for fast lookup, we can at least order the tests from most likely (guessing of course) to least likely.

e.g., is Unknown really the most likely case? I hope not.

Reply:
reordered